templates/index.html

```diff
@@ -70,40 +87,40 @@
     <h1 class="ui header">
       Tracking Demo
     </h1>
-
-    <p>Drag and drop or click to upload a video file for analysis.</p>
+    <ol>
+      <li>
+        Click <a href="/static/sample_video.mp4" download>here</a> to download a sample video.
+      </li>
+      <li>
+        Drag & drop the downloaded video into the rectangle below (or click on the rectangle to upload).
+      </li>
+      <li>
+        Wait until the video has been processed. Progress information will be posted to the screen.
+      </li>
+    </ol>	
 
     <div id="dropzone">
       <form class="dropzone needsclick" id="file-upload" action="/upload">
         <div class="dz-message needsclick">
           Drop files here or click to upload.
         </div>
       </form>
-    </div>
-
-    {% if jobs %}
+    </div>    
+    {% set session_is_empty = session|empty_session %}
+    {% if not session_is_empty %}
       <h3>Videos</h3>
-      <p>Refresh the page to update.</p>
-      <ul>
-        {% for job in jobs %}
-          <li>
-            <ul>
-              <li>
-                filename: {{ job.filename }}
-              </li>
-              <li>
-                is_finished: {{ job.is_finished }}
-              </li>
+        {% for job_dict in session['jobs'] %}
+	<ul style="list-style-type:none;">
               <li>
-                result: {{ job.result }}
+              {{ job_dict | refresh_job }}
```

Comment:
abrichr 9 days ago Member
Is this necessary?

Reply:
Yes, jobs don't get refreshed after a page reload, they need to be explicitly refreshed

Reply:
Why are you doing it in the template?

Reply:
https://github.com/MLDSAI/VisionTracing/pull/2/commits/82f05c905d83cd1a95aa8d8e52a7e97d288f99cb

I changed the name from refresh_job to get_filename cause that is more indicative of what the function is doing. When the client issues a new http request, the server renders the template (and in the process, refreshes any active jobs and displays their filenames). This is specifically for when the user refreshes a page. All of the display messages that do not require refreshing use sockets, and are completely independent of the job meta dictionary.